### DISCUSSION TITLE
Multiple repositories

### DISCUSSION THREAD

**Discussion by gogobd** [coder/code-server#3108] [Q&A]
Hello, every 01!

Is it possible with cdr/code-server to open multiple subdirectories or add multiple folders to the workspace where each folder contains a separate git repository? VSCS is able to do this but I'm only using code server; I found that with VSCS one can do "code ." in each repository to add it to the "Source Control Repositories" section.

Is this possible with code-server?

**Comment by jsjoeio** [comment]
I'm probably missing something from your question, but based on how I'm interpreting it, this should work.

In my demo, `/dev` contains various folders, each of which is a separate git repository. You can see when I open the Source Control panel that code-server recognizes changes in each git folder.

## Demo
https://user-images.githubusercontent.com/3806031/114447963-d583b180-9b87-11eb-8986-a9a77b55278a.mov

Let me know what I'm missing and we'll see how we can help!

**Reply by gogobd** [reply]
Thank you for your response, this is exactly what I was looking for! Unfortunately I can't get my vscs to pick up .git repos in subprojects; an example file structure I'm using is:

```
baseproject
├── subprojects
│   ├── subproject1
│   ├── subproject2
│   ├── subproject3
│   ├── subproject4
│   ├── subproject5
│   ├── ...
└── src
    └── ...
```

Is there a setting that I am maybe missing, or some way I can make code server explicitly aware of the subprojects? Is it possible that in some way .gitignore could prevent code server from picking up the subprojects?

**Reply by gogobd** [reply]
I was trying to toggle "git.autoRepositoryDetection": "subFolders" but that didn't help :-/

Even adding one of the subproject's paths to "git.scanRepositories": [] wasn't successful.

**Reply by jsjoeio** [reply]
> Is it possible that in some way .gitignore could prevent code server from picking up the subprojects?

There could be some funkiness happening with the `.gitignore` 🤔 Hmm...have you looked at these threads by chance:

- [VS Code not recognizing a git repo workspace](https://github.com/microsoft/vscode/issues/2947)
- [VS Code Not Recognising Git Repo](https://github.com/microsoft/vscode/issues/97392)
- [Git missing in VS Code – No source control providers](https://stackoverflow.com/questions/46609255/git-missing-in-vs-code-no-source-control-providers)

**Reply by gogobd** [reply]
Still no dice.

my git directories are

```
baseproject/.git
baseproject/subprojects/subproject1/.git
baseproject/subprojects/subproject2/.git
baseproject/subprojects/subproject3/.git
...
```

I open the folder "baseproject" in vscs.

**Reply by jsjoeio** [reply]
Hmm...do you want to post reproduction steps and I'll try reproducing locally? Here's a template from our [bug report](https://github.com/cdr/code-server/issues/new?assignees=&labels=&template=bug-report.md&title=)

**Comment by gogobd** [comment]
I issued #3321
